src/VisualStudio/Core/Def/ExternalAccess/ProjectSystem/Api/IProjectSystemUpdateReferenceOperation.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.LanguageServices.ExternalAccess.ProjectSystem.Api
+{
+    internal interface IProjectSystemUpdateReferenceOperation
+    {
+        /// <summary>
+        /// Applies a reference update operation to the project file.
+        /// </summary>
+        /// <returns>A boolean indicating success.</returns>
+        Task<bool> ApplyAsync();
+
+        /// <summary>
+        /// Reverts a reference update operation to the project file.
+        /// </summary>
+        /// <returns>A boolean indicating success.</returns>
+        Task<bool> RevertAsync();
```

Comment:
What is the behavior of calling "RevertAsync" without ever calling Apply? Maybe this should be `Task<IProjectSystemUpdateRevertOperation> ApplyAsync()`

Reply:
Similarly, what is the behavior of calling `ApplyAsync` multiple times?

Reply:
How would you feel about CanRevert & CanApply properties?

Reply:
I think it's a bit overkill. Instead outlining the behavior in comments here is probably fine. I.E: If an exception is thrown because it has already been applied/hasn't been applied. That way implementers and callers both know the contract